src/Elastic.Apm/Helpers/StacktraceHelper.cs

```diff
@@ -64,7 +64,7 @@ internal static class StacktraceHelper
 
 					var fileName = frame?.GetFileName();
 
-					logger.Trace()?.Log("{MethodName}, {lineNo}", functionName, frame?.GetFileLineNumber());
+					logger?.Trace()?.Log("{MethodName}, {lineNo}", functionName, frame?.GetFileLineNumber());
```

Comment:
The null conditional operator on `logger` looks unneccessary. `StacktraceHelper` is internal and `GenerateApmStackTrace` is called only by agent code where the logger should never be null (if it is null, it's a bug).